internal/core/command/put_test.go

```diff
@@ -73,3 +80,9 @@ func TestNewPutCommandBody(t *testing.T) {
 		t.Error("Failed to verify the request body contents")
 	}
 }
+
+func newRequestWithContentType(contentType string) *http.Request {
```

Comment:
Even though we are now including a `Content-type` in the requests, we are never verifying the resulting request has the correct headers. Can we add these tests to ensure the newly added functionality?